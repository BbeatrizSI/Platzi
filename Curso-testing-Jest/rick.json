{
    "id": 1,
    "name": "Rick SÃ¡nchez",
    "status": "Alive",
    "species": "Human",
    "gender": "Male"
}