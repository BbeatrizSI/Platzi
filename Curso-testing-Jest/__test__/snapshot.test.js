import {getCharacter} from '../snapshot';
import  rick from '../rick.json';

describe('Es hora de las instant치neas', () => {
    test('Snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });
    test('Siempre fallar치 el Snapshot', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20)
        }
        expect(user).toMatchSnapshot();
    });
    test('Permitimos una excepci칩n', () => {
        const user2 = {
            name: 'B치rbara',
            id: Math.floor(Math.random() * 20)
        }
        expect(user2).toMatchSnapshot({
            id: expect.any(Number)
        });

    })
});