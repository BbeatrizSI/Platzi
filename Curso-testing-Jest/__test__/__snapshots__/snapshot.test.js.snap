// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantáneas Permitimos una excepción 1`] = `
Object {
  "id": Any<Number>,
  "name": "Bárbara",
}
`;

exports[`Es hora de las instantáneas Siempre fallará el Snapshot 1`] = `
Object {
  "createAt": 2020-11-06T11:39:39.774Z,
  "id": 0,
}
`;

exports[`Es hora de las instantáneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sánchez",
  "species": "Human",
  "status": "Alive",
}
`;
